acvm-repo/acvm/src/pwg/blackbox/hash.rs

```diff
@@ -149,6 +149,16 @@ pub(crate) fn solve_poseidon2_permutation_opcode(
             ),
         ));
     }
+    if len as usize != outputs.len() {
+        return Err(OpcodeResolutionError::BlackBoxFunctionFailed(
+            acir::BlackBoxFunc::Poseidon2Permutation,
+            format!(
+                "the number of outputs does not match specified length. {} > {}",
```

Comment:
```suggestion
                "the number of outputs does not match specified length. {} != {}",
```

The number of outputs could also be less than the number of inputs, no?

Reply:
This also applies to the other error.